platform/ios/src/MGLMapView.mm

```diff
@@ -3550,8 +3550,12 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
         {
             id <MGLAnnotation> annotation = [self annotationWithTag:annotationTag];
             NSAssert(annotation, @"Unknown annotation found nearby tap");
```

Comment:
Just noting that this assertion will fail in a debug build if you perform the reproduction steps in https://github.com/mapbox/mapbox-gl-native/issues/8163#issuecomment-291695262. The issue is that mbgl notion of visible annotations will always lag a bit behind the platforms' notion of the same when annotations are removed.